primitives/block/src/equivocation_proof.rs

```diff
@@ -327,20 +331,30 @@ impl DoubleProposalProof {
     pub fn new(
         validator_address: Address,
         mut header1: MacroHeader,
+        mut round1: u32,
+        mut valid_round1: Option<u32>,
         mut justification1: SchnorrSignature,
         mut header2: MacroHeader,
+        mut round2: u32,
+        mut valid_round2: Option<u32>,
```

Comment:
Should I merge the two `round` parameters into one, given that they have to be the same?

Reply:
I would prefer these be `TendermintProposal`, making the signature much smaller. If that turns out to be annoying or to require unnecessary cloning, then I think one round could be removed yes.